### PR TITLE
feat(insights): add analytics to insight modules [part 1]

### PR DESCRIPTION
**Pull request by KevinL10**
Adds interaction analytics to insight modules. This PR includes:
- App Starts
- Assets
- Screen Loads
- action/domain selectors

The next PR will include:
- general (search, API requests, charts)
- web vitals

### CODE REVIEW

File: static/app/utils/analytics/insightAnalyticEvents.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+export type InsightEventParameters = {
+  'insight.app_start.select_start_type': {type: string};
+  'insight.app_start.spans.filter_by_device_class': {filter: string};
+  'insight.app_start.spans.filter_by_operation': {filter: string};
+  'insight.app_start.spans.toggle_sample_type': {type: string};
+  'insight.asset.filter_by_blocking': {filter: string};
+  // Don't specify filter because page filter are arbitrary values
+  'insight.asset.filter_by_page': {};
+  'insight.asset.filter_by_type': {filter: string};
```
Comment (DominikB2014) [inline static/app/utils/analytics/insightAnalyticEvents.tsx:9]: Not, but this should probably match the module name, which is Assets (plural)

File: static/app/views/performance/mobile/components/deviceClassSelector.tsx [hunk 2]
```diff
@@ -2,18 +2,27 @@ import type {ComponentProps} from 'react';
...
   clearSpansTableCursor?: boolean;
+  moduleName?: ModuleName;
   size?: ComponentProps<typeof CompactSelect>['size'];
 }
 
-export function DeviceClassSelector({size = 'xs', clearSpansTableCursor}: Props) {
+export function DeviceClassSelector({
+  size = 'xs',
+  clearSpansTableCursor,
+  moduleName = ModuleName.OTHER,
```
Comment (DominikB2014) [inline static/app/views/performance/mobile/components/deviceClassSelector.tsx:22]: Is there a case that we need to default to Other?

File: static/app/views/performance/mobile/components/deviceClassSelector.tsx [hunk 3]
```diff
@@ -32,6 +41,17 @@ export function DeviceClassSelector({size = 'xs', clearSpansTableCursor}: Props)
...
+          trackAnalytics('insight.app_start.spans.filter_by_device_class', {
+            organization,
+            filter: newValue.value,
+          });
+        } else if (moduleName === ModuleName.SCREEN_LOAD) {
+          trackAnalytics('insight.screen_load.spans.filter_by_device_class', {
+            organization,
+            filter: newValue.value,
+          });
+        }
```
Comment (DominikB2014) [inline static/app/views/performance/mobile/components/deviceClassSelector.tsx:54]: Could we dynamically set this?
Ie `insight.${moduleName}...`

**Review by gggritso — APPROVED** [review]
👍🏻

**Review by DominikB2014 — APPROVED** [review]
Left some bits, but lgtm

File: static/app/views/performance/mobile/components/deviceClassSelector.tsx (hunk 3 above)
Comment (KevinL10) [reply static/app/views/performance/mobile/components/deviceClassSelector.tsx:54]: moduleName can also be `mobile-ui` which we do not track, so I explicitly included these cases

File: static/app/utils/analytics/insightAnalyticEvents.tsx (hunk 1 above)
Comment (KevinL10) [reply static/app/utils/analytics/insightAnalyticEvents.tsx:9]: thanks, will fix in the next

File: static/app/views/performance/mobile/components/deviceClassSelector.tsx (hunk 2 above)
Comment (KevinL10) [reply static/app/views/performance/mobile/components/deviceClassSelector.tsx:22]: for `mobile-ui`

**Pull request merged**